conans/model/cpp_package.py

```diff
@@ -0,0 +1,71 @@
+import json
+from conans.util.files import load, save
+
+class CppPackage(object):
+    """
+    models conan_package.json, serializable object
+    """
+
+    DEFAULT_FILENAME = "cpp_package.json"
+
+    def __init__(self):
+        self.components = dict()
+
+    @classmethod
+    def load(cls, filename=DEFAULT_FILENAME):
+        """
+        loads package model into memory from the conan_package.json file
+        :param filename: path to the cpp_package.json
+        :return: a new instance of CppPackage
+        """
+        def from_json(o):
+            if "components" in o:
+                return {n: CppPackage.Component(c) for n, c in o["components"].items()}
+            else:
+                return o
+        conan_package = CppPackage()
+        conan_package.components = json.loads(load(filename), object_hook=from_json)
+        return conan_package
+
+    def save(self, filename=DEFAULT_FILENAME):
+        """
+        saves package model from memory into the cpp_package.json file
+        :param filename: path to the cpp_package.json
+        :return: None
+        """
+        save(filename, json.dumps(self, default=lambda o: o.__dict__))
+
+    def package_info(self, conanfile):
+        """
+        performs an automatically generated package_info method on conanfile, populating
+        conanfile.package_info with the information available inside cpp_package.json
+        :return: None
+        """
+        for cname, component in self.components.items():
+            conanfile.cpp_info.components[cname].libs = component.libs
+            conanfile.cpp_info.components[cname].requires = component.requires
+            for generator, gname in component.names.items():
+                conanfile.cpp_info.components[cname].names[generator] = gname
```

Comment:
``names`` no longer the way to define these things, better user ``set_property()``.

The UI might need to evolve, like it would be better to return a ``NewCppInfo`` information instead, maybe? We can leave it from now, and see how it evolves.

Isn't it possible to get the ``include`` directories from the API? I would expect to obtain:

- Includedirs
- Libdirs
- libraries
- Component names

Reply:
> names no longer the way to define these things, better user set_property().

not sure I understand, `set_property`, do you mean CMake [thing](https://cmake.org/cmake/help/latest/command/set_property.html)? I don't see how can I set it from within `CppInfo`. do you have an example?

> The UI might need to evolve, like it would be better to return a NewCppInfo information instead, maybe? We can leave it from now, and see how it evolves.

from the functional perspective yes, this makes `CppInfo` an immutable object, and `package_info` will be pure function, which should result in several benefits

> Isn't it possible to get the include directories from the API?

unfortunately, I don't see anything related to the `includedirs` in the current response.
still, I suppose, having some information auto-generated is much better than having no information at all. author always can manually fill missing `CppInfo` fields, as you have stated previously.
for the most of cases, it won't be ever needed, as for standard layout `include`/`lib` directories are added automatically.